rir/src/compiler/translations/pir_2_rir.cpp

```diff
@@ -946,32 +1032,30 @@ size_t Pir2Rir::compile(Context& ctx, Code* code) {
                 break;
             }
             case Tag::CallSafeBuiltin: {
-                assert(false && "not yet implemented.");
+                auto blt = CallSafeBuiltin::Cast(instr);
+                // no environment
+                blt->eachArg([&](Value* arg) { load(it, arg); });
+                cs.insertStackCall(Opcode::static_call_stack_, blt->nargs(), {},
+                                   Pool::get(blt->srcIdx), blt->blt);
+                store(it, blt);
                 break;
             }
             case Tag::MkEnv: {
                 auto mkenv = MkEnv::Cast(instr);
                 loadEnv(it, mkenv->parent());
-                cs << BC::makeEnv() << BC::dup() << BC::setEnv();
+                cs << BC::makeEnv() << BC::setEnv();
                 currentEnv = mkenv;
                 // bind all args
                 // TODO: maybe later have a separate instruction for this?
                 mkenv->eachLocalVar([&](SEXP name, Value* val) {
                     load(it, val);
                     cs << BC::stvar(name);
                 });
+                cs << BC::getEnv();
                 store(it, mkenv);
```

Comment:
can't you do the store before you set all variables

Reply:
Good idea, thanks!
Question, though, do you think this should restore the old environment, or leave the newly created one as current env?